Rework

Silumgaaar opened the pull request.

Silumgaaar committed:
rew

Silumgaaar committed:
Создание объектов словаря

Добавил код создания объектов словаря. В основной код пока не добавлял изменения

Silumgaaar committed:
rework

Silumgaaar committed:
rework ConsoleConstains

Раскидал константы по местам их использования

NikolayKlyatishev commented on src/main/view/ConsoleApp.java:
```diff
@@ -0,0 +1,90 @@
+public class ConsoleApp {
+
+    private static final String EXIT_INFO = "Type Exit to end the program ";
+    private static final String EXIT = "Exit";
```
Стоит инкапсулировать типы команд. Все одном месте.

Silumgaaar committed:
enum

Теперь информация о команде хранится в enum

Silumgaaar committed:
rework ConsoleApp

Переделал класс ConsoleApp убрал команду Start, переделал команду Back; Команды теперь вызываются по цифре.

NikolayKlyatishev commented on src/main/view/ConsoleApp.java:
```diff
@@ -0,0 +1,47 @@
+
+    public void start(){
+        String userChoice = "";
+        command = commands.get(5);
```
Почему по цифре? С ходу не ясно к какой команде относится цифра 5. Это неудобно читать и сложно поддерживаить.

NikolayKlyatishev commented on src/main/view/ConsoleApp.java:
```diff
@@ -0,0 +1,47 @@
+    private static final String COMMAND_NOT_FOUND = "Command entered not found ";
+    private static final String ENTER_COMMAND = "Enter a command: ";
+    private static final String NUMBER_FORMAT_EXCEPTION = "Enter the number";
+    HashMap<Integer, Commands> commands = new HashMap<>();
```
Тут нарушение инкапсуляции состояния объекта. Поля должны быть скрыты.

NikolayKlyatishev commented on src/main/view/ConsoleApp.java:
```diff
@@ -0,0 +1,47 @@
+public class ConsoleApp {
+
+    private static final String EXIT_INFO = "Type Exit to end the program ";
+    private static final String EXIT = "Exit";
```
Почему одна из команд все еще является состоянием приложения?

NikolayKlyatishev commented on src/main/view/ConsoleApp.java:
```diff
@@ -0,0 +1,47 @@
+        System.out.println(EXIT_INFO);
+        while (!userChoice.equalsIgnoreCase(EXIT)){
+            System.out.println(ENTER_COMMAND);
+            userChoice = User.choice.next();
```
Думаю не стоит сканнер прятать в отдельный класс.

NikolayKlyatishev commented on src/main/dictionarywork/Dictionary.java:
```diff
@@ -0,0 +1,160 @@
+        return dictionary.get(key);
+    }
+
+    private String getName(){
```
Для чего нужен приватный "геттер"? Почему не использовать поле?

NikolayKlyatishev commented on src/main/dictionarywork/DictionaryManager.java:
```diff
@@ -0,0 +1,16 @@
+public interface DictionaryManager {
+
+    HashMap<String,String> getDictionary();
+    List<String> getDictionaries();
```
Методы getDictionaries newDictionary нарушают принцип единой ответственности. Назначение данного интерфейса это просто crud операции. Добавленные же методы отвечают за некоторую бизнес логику.

NikolayKlyatishev commented on src/main/dictionarywork/DictionaryManager.java:
```diff
@@ -0,0 +1,16 @@
+
+public interface DictionaryManager {
+
+    HashMap<String,String> getDictionary();
```
Стоит изменить наименование. Например gelAll

NikolayKlyatishev commented on src/main/dictionarywork/DictionaryManager.java:
```diff
@@ -0,0 +1,16 @@
+
+    HashMap<String,String> getDictionary();
+    List<String> getDictionaries();
+    boolean newDictionary(String name);
```
Мне кажется инициализация объекта не относится к поведению. Думаю это нарушение принципа единой ответственности. Возможно тут нужна просто еще одна реализация. Очень хороший вариант написать реализацию отдельного класса для обработки выбора словарей. Потом проверь - есть ли общее поведение. Может можно объединить их интерфейсом.

NikolayKlyatishev commented on src/main/view/InfoCommands.java:
```diff
@@ -0,0 +1,51 @@
+package main.view;
+
+public enum InfoCommands {
```
Стоит оставить наименование Commands.  Интерфейс же - это действие. Выбери для интерфейса более подходящее имя.

NikolayKlyatishev commented on src/resources/DictionaryType1.txt:
```diff
@@ -0,0 +1,8 @@
+qpow-прив
+qwer-йцук
```
Почему сам словарь у тебя в индексе гита?
